network/processor.go

```diff
@@ -89,7 +89,13 @@ Loop:
 			output <- &Ping{}
 		case <-timeout.C:
 			log.Info().Msg("peer timed out, dropping")
-			peers.Drop(address)
+			err := peers.Drop(address)
+			if err != nil {
+				log.Error().Err(err).Msg("could not drop peer")
+			} else {
```

Comment:
Better to use a continue and no else. Please read the Go style guide again to refresh it a bit. Go always tries to keep statement on the lowest level of indentation if it's the success/default case.